com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StepsUtil.java

```diff
@@ -345,13 +345,19 @@ static void setAppsToUndeploy(VariableScope scope, List<CloudApplication> apps)
     }
 
     public static List<String> getServicesToDelete(VariableScope scope) {
-        TypeReference<List<String>> type = new TypeReference<List<String>>() {
-        };
-        return getFromJsonBinary(scope, Constants.VAR_SERVICES_TO_DELETE, type);
+        return getObject(scope, Constants.VAR_SERVICES_TO_DELETE, Collections.emptyList());
     }
 
     public static void setServicesToDelete(VariableScope scope, List<String> services) {
-        setAsJsonBinary(scope, Constants.VAR_SERVICES_TO_DELETE, services);
+        scope.setVariable(Constants.VAR_SERVICES_TO_DELETE, services);

```

Comment:
this was a json binary before, are you sure the list will be properly serialized now with just setVariable in all cases?

Reply:
This change was necessary because Flowable engine throw exception that result from getServicesToDelete() method is not a Collection.

Reply:
From the conversation, my understanding is that you'd want to use this variable in flowable bpmn diagram in order to parallelize the deletion of services, and you could not do that because you need the variable not to be in 'bytearray' format in the database. This is okay. However - is this backwards compatible? What would happen if I retry my deployment, that has this variable stored as 'bytearray' after the release of the diagram change happens - will the deployment fail?